openreview/conference/templates/programchairWebfield.js

```diff
@@ -959,31 +959,53 @@ var displayStatsAndConfiguration = function(conferenceStats) {
     if (invitationMap[SENIOR_AREA_CHAIRS_ID + '/-/Assignment_Configuration']) {
-      html += '<li><a href="/assignments?group=' + SENIOR_AREA_CHAIRS_ID + '&referrer=' + referrerUrl + '">Senior Area Chairs Paper Assignment</a> open until Reviewing starts</li>';
+      not_dated_elements.push('<li><a href="/assignments?group=' + SENIOR_AREA_CHAIRS_ID + '&referrer=' + referrerUrl + '">Senior Area Chairs Paper Assignment</a> open until Reviewing starts</li>')
     }
   }
   if (AREA_CHAIRS_ID) {
-    html += renderInvitation(invitationMap, AREA_CHAIRS_ID + '/-/' + BID_NAME, 'Area Chairs Bidding')
-    html += renderInvitation(invitationMap, REVIEWERS_ID + '/-/Recommendation', 'Reviewer Recommendation')
+    push_to_dated_arrays(invitationMap, AREA_CHAIRS_ID + '/-/' + BID_NAME, 'Area Chairs Bidding', dated_elements, not_dated_elements)
+    push_to_dated_arrays(invitationMap, REVIEWERS_ID + '/-/Recommendation', 'Reviewer Recommendation', dated_elements, not_dated_elements)
     if (invitationMap[AREA_CHAIRS_ID + '/-/Assignment_Configuration']) {
-      html += '<li><a href="/assignments?group=' + AREA_CHAIRS_ID + '&referrer=' + referrerUrl + '">Area Chairs Paper Assignment</a> open until Reviewing starts</li>';
+      not_dated_elements.push('<li><a href="/assignments?group=' + AREA_CHAIRS_ID + '&referrer=' + referrerUrl + '">Area Chairs Paper Assignment</a> open until Reviewing starts</li>')
     }
   }
   if (invitationMap[REVIEWERS_ID + '/-/Assignment_Configuration']) {
-    html += '<li><a href="/assignments?group=' + REVIEWERS_ID + '&referrer=' + referrerUrl + '">Reviewers Paper Assignment</a> open until Reviewing starts</li>';
+    not_dated_elements.push('<li><a href="/assignments?group=' + REVIEWERS_ID + '&referrer=' + referrerUrl + '">Reviewers Paper Assignment</a> open until Reviewing starts</li>')
   }
-  html += renderInvitation(invitationMap, CONFERENCE_ID + '/-/' + OFFICIAL_REVIEW_NAME, 'Reviewing')
-  html += renderInvitation(invitationMap, CONFERENCE_ID + '/-/' + COMMENT_NAME, 'Commenting')
-  html += renderInvitation(invitationMap, CONFERENCE_ID + '/-/' + OFFICIAL_META_REVIEW_NAME, 'Meta Reviews')
-  html += renderInvitation(invitationMap, CONFERENCE_ID + '/-/' + DECISION_NAME, 'Decisions')
+  push_to_dated_arrays(invitationMap, CONFERENCE_ID + '/-/' + OFFICIAL_REVIEW_NAME, 'Reviewing', dated_elements, not_dated_elements)
+  push_to_dated_arrays(invitationMap, CONFERENCE_ID + '/-/' + COMMENT_NAME, 'Commenting', dated_elements, not_dated_elements)
+  push_to_dated_arrays(invitationMap, CONFERENCE_ID + '/-/' + OFFICIAL_META_REVIEW_NAME, 'Meta Reviews', dated_elements, not_dated_elements)
+  push_to_dated_arrays(invitationMap, CONFERENCE_ID + '/-/' + DECISION_NAME, 'Decisions', dated_elements, not_dated_elements)
+
+  // Sort + append in dated order, followed by not-dated content
+  dated_elements.sort(function (a, b) {return a['date'] - b['date']})
+
+  for (var i = 0; i < dated_elements.length; i++)
```

Comment:
Generally, we prefer map and filter for this type of array manipulation. This could be written as:

```js
html += datedElements.map(function(elem) { return elem.content; }).join('\n');
html += undatedElements.join('\n');
```

Reply:
Alright, I've replaced the two for loops with the `map()` and `join()` functions. I wasn't aware that JavaScript had map and reduce functions for generic arrays, that's cool